From 4e7e0f40d12ba8febd5fcd9e8e604def8bf56d8d Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 2 Dec 2020 02:16:16 +0100
Subject: [PATCH 3/4] Fix out-of-source-tree CXX build failures

Lack of CXXFLAGS here. I would expect that CPPFLAGS would
have made it through but that seems not to be the case.

Anyway it seems like being able to specify global CXXFLAGS
in one place is useful and this achieves that too.
---
 src/Makefile.am | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 2d8ba6b..db6b531 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -157,13 +157,13 @@ endif
 if HAVE_OPENCV
 plugin_LTLIBRARIES += facebl0r.la
 facebl0r_la_SOURCES = filter/facebl0r/facebl0r.cpp
-facebl0r_la_CFLAGS = @OPENCV_CFLAGS@ @CFLAGS@ 
-facebl0r_la_CXXFLAGS = @OPENCV_CFLAGS@ @CFLAGS@
+facebl0r_la_CFLAGS = @OPENCV_CFLAGS@ @CFLAGS@
+facebl0r_la_CXXFLAGS = @OPENCV_CFLAGS@ @CXXFLAGS@
 facebl0r_la_LIBADD = @OPENCV_LIBS@
 plugin_LTLIBRARIES += facedetect.la
 facedetect_la_SOURCES = filter/facedetect/facedetect.cpp
 facedetect_la_CFLAGS = @OPENCV_CFLAGS@ @CFLAGS@
-facedetect_la_CXXFLAGS = @OPENCV_CFLAGS@ @CFLAGS@
+facedetect_la_CXXFLAGS = @OPENCV_CFLAGS@ @CXXFLAGS@
 facedetect_la_LIBADD = @OPENCV_LIBS@
 endif
 
@@ -335,6 +335,7 @@ xfade0r_la_SOURCES = mixer2/xfade0r/xfade0r.cpp
 
 AM_CPPFLAGS = -I@top_srcdir@/include -Waddress -Wtype-limits -Wsign-compare
 AM_CFLAGS = -I@top_srcdir@/include -Waddress -Wtype-limits -Wsign-compare
+AM_CXXFLAGS = -I@top_srcdir@/include -Waddress -Wtype-limits -Wsign-compare
 AM_LDFLAGS = -module -avoid-version -lm -export-dynamic
 AM_LIBTOOLFLAGS = --tag=disable-static
 
-- 
2.23.0

